searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "tensor_buffer_operations.h"
+#include <vespa/eval/eval/fast_value.hpp>
+#include <vespa/eval/eval/value.h>
+#include <vespa/eval/eval/value_codec.h>
+#include <vespa/eval/eval/value_type.h>
+#include <vespa/eval/streamed/streamed_value_view.h>
+#include <vespa/vespalib/util/arrayref.h>
+#include <vespa/vespalib/util/shared_string_repo.h>
+#include <algorithm>
+
+using vespalib::ArrayRef;
+using vespalib::ConstArrayRef;
+using vespalib::MemoryUsage;
+using vespalib::SharedStringRepo;
+using vespalib::StringIdVector;
+using vespalib::eval::FastAddrMap;
+using vespalib::eval::FastValueIndex;
+using vespalib::eval::StreamedValueView;
+using vespalib::eval::TypedCells;
+using vespalib::eval::Value;
+using vespalib::eval::ValueType;
+using vespalib::eval::self_memory_usage;
+using vespalib::string_id;
+
+namespace search::tensor {
+
+namespace {
+
+uint32_t
+adjust_min_alignment(size_t min_alignment)
+{
+    // Also apply alignment for num_subspaces and labels
+    return std::max(std::max(sizeof(uint32_t), sizeof(string_id)), min_alignment);
+}
+
+struct MyFastValueView final : Value {
```

Comment:
Would just _FastValueView_ be sufficient here?

Reply:
Renamed.